### PR TITLE
Ports rebindable locking movement key and move intent toggle hotkey from /tg/ again again

### PR DESCRIPTION
**Pull request by BrumalLake**
## About The Pull Request
https://github.com/BeeStation/BeeStation-Hornet/pull/4460
#4775
Ports https://github.com/tgstation/tgstation/pull/53771
Ports https://github.com/tgstation/tgstation/pull/47569
Makes the locking movement to turn direction key rebindable (by default Ctrl)
Adds a new hotkey that lets you toggle walking (unbound by default)
Note: You still have to rebind your look direction hotkeys individually along with the locking movement hotkey or you won't turn

review this time please

## Why It's Good For The Game
no longer have to go to the corner of your screen and click to toggle walking in the middle of combat to block
locking movement is more free and can be put where you want

## Changelog
:cl:
add: Added hotkey to toggle walking/running
add: Ability to rebind locking movement hotkey
/:cl:

### CODE REVIEW

File: code/datums/keybinding/mob.dm [hunk 1]
```diff
@@ -226,3 +240,21 @@
 	if(!user.mob) return
 	user.body_l_leg()
 	return TRUE
+
+/datum/keybinding/mob/prevent_movement
+	key = "Ctrl"
+	name = "block_movement"
+	full_name = "Block movement"
+	description = "Prevents you from moving"
```
Comment (francinum) [inline code/datums/keybinding/mob.dm]: Needs clarification. The var isn't much better either. This seems like it *disables* movement period.

Comment (BrumalLake) [reply code/datums/keybinding/mob.dm]: that's exactly what it does
you have to rebind your face direction keys with it

Comment (francinum) [reply code/datums/keybinding/mob.dm]: ...as in, it prevents *all* movement._

Comment (BrumalLake) [reply code/datums/keybinding/mob.dm]: you mean, you can't get pushed around pulled or anything?

**Comment by qwertyquerty** [comment]
Tested?

**Comment by BrumalLake** [comment]
yeah i tested it you just gotta change your face direction key binds with it to turn and the toggle walk works
the description before was perfectly accurate, all it needed maybe was to say that you had to bind your face keys along with it

**Review by qwertyquerty — APPROVED** [review]

**Pull request merged**